### DISCUSSION TITLE
Alternative to Inquirer Prompt for Multi-line Text Input

### DISCUSSION THREAD

**Discussion by kyletechie** [SBoudrias/Inquirer.js#1583] [Q&A] [answered]
Is there an alternative to the Inquirer prompt that supports multi-line text input, without requiring the editor type option? I'm looking for a simple input prompt.

For example, if the prompt type is set to input, and I paste a multi-line text like:

```
this is
multi line
text
```

the prompt only captures "this is" and stops there, instead of recognizing the full text:

```
this is
multi line
text
```

**Comment by SBoudrias** [comment, marked as answer]
Hey @Kairu-bit, I'd really like the input prompt to allow inline multiline. But I think we're limited by how readline work.

If you look at the [Node.js documentation example](https://nodejs.org/api/readline.html):

```js
import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

const rl = readline.createInterface({ input, output });

const answer = await rl.question('What do you think of Node.js? ');

console.log(`Thank you for your valuable feedback: ${answer}`);

rl.close();
```

You'll see any press of `enter` will trigger the line event.

I'd be happy to work around this if we find a clever way.

**Comment by SBoudrias** [comment]
Interestingly, I've seen this today in a Go library. It'd be sick to have a similar version within Inquirer https://github.com/charmbracelet/bubbles#text-area

**Comment by abhinab-choudhury** [comment]
```js
process.stdin.setRawMode(true);
```
This is a good idea, but adding support for special characters will take extra effort (\u0003 for Ctrl+C, \r for Enter, etc.).

**Comment by prassamin** [comment]
Hey! You can try this one. I actually built it recently for personal use. It handles multiline input exactly like you’re asking for.
[https://github.com/PRASSamin/cli-prompts/](https://github.com/PRASSamin/cli-prompts/)
